eng/generators.targets

```diff
@@ -16,17 +16,12 @@
                         and '$(IsSourceProject)' == 'true'
                         and '$(MSBuildProjectExtension)' == '.csproj'
                         and (
-                          ('@(Reference)' != ''
+                          '$(TargetFrameworkMoniker)' != '$(NetCoreAppCurrentTargetFrameworkMoniker)'
+                          or '$(DisableImplicitFrameworkReferences)' != 'true'
```

Comment:
NIT: can we please update the comment above to explain the condition in more detail given it is a complex one?

Reply:
We can simplify these conditions vastly and I originally had that change in this PR but then moved it out as it's unrelated to my change. I will submit a follow-up PR that does the clean-up and add a comment.